pkg/controllers/provisioning/v1alpha1/reallocation/terminate.go

```diff
@@ -130,8 +130,6 @@ func (t *Terminator) deleteNodes(ctx context.Context, nodes []*v1.Node, provisio
 		return fmt.Errorf("terminating cloudprovider instance, %w", err)
 	}
 	// 2. Delete node in APIServer
-	// TODO: Prevent leaked nodes: ensure a node is not deleted in apiserver if not deleted in cloudprovider
```

Comment:
We can probably remove this todo anyway, but before I forget, does the `Delete` implementation (for AWS) correctly handle the case where it tries to delete an already-deleted node? (It should recognize this case and not consider it an error). Otherwise you can end up in a situation where the nodes never get deleted on the K8s side (i.e., the `kubeClient.Delete` will never get called because of 1 (or more) seemingly un-deletable node)?

Reply:
There's a safety valve in the Kubernetes cloud controller manager that checks EC2 if instances exist and deletes the node if it doesn't. a.k.a., kubernetes will reconcile.

Reply:
Do we need to do this at all then? I still think there would be a case where this code never gets past the error (unless we also check whether or not k8s thinks the node exists?)

Reply:
Deleting the node in APIServer rather than waiting for garbage collection allows the node to get cleaned up immediately. Performance wise, the benefit is that it won't get called to delete in further reconcile loops (which if a reconcile loop is 5s and the garbage collection is 300s, that's 60 unnecessary calls). 
Functionality wise, I'm unsure if the deletion API for AWS correctly handles Jacob's aforementioned case. If it doesn't properly handle it, then we will be continually returning an error here until garbage collection cleans it up.

Reply:
Ok, I cut a ticket (https://github.com/awslabs/karpenter/issues/340) to investigate, and will approve this change.